### ISSUE TITLE
ComboBox in a ScrollViewer in Xaml Islands can be scrolled off the window

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/microsoft-ui-xaml#4554]
**Steps to reproduce the bug**
In a XAML Islands application, create a `ComboBox` in a `ScrollViewer`:

```xaml
<Page>
    <ScrollViewer>
        <StackPanel>
            <ComboBox>
                <ComboBoxItem Content="Foo" />
                <ComboBoxItem Content="Bar" />
                <ComboBoxItem Content="Baz" />
            </ComboBox>
        </StackPanel>
    <ScrollViewer>
</Page>
```

(The exact one is over at https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalSettingsEditor/Launch.xaml#L38-L67)

Run the app, open the comboBox, then mouse over the rest of the scroll viewer and start scrolling.

![gh-9320-000](https://user-images.githubusercontent.com/18356694/111514586-e1fb2280-871f-11eb-93bf-bb08d0e4fd16.gif)

**Expected behavior**
In a non-islands app, you can't scroll the scrollviewer while the combobox is open. You have to dismiss the box first.

**Actual behavior**

The combo box's entries float off the top of the window into space

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml.2.5.0-prerelease.201202003]

I'm on 21337.rs_wdx_dxp_sh

**Comment by DHowett** [comment]
This is amazing.

**Comment by StephenLPeters** [comment]
@RBrid looks like in a UWP app we disable scrolling while a light dismiss popup is open, likely via the lightdismiss layer that fills the entire screen.  I'm guessing that in an islands scenario that light dismiss layer might not fill the entire screen. which could allow scrolling? Do you concur?

@JesseCol for the islands perspective.

**Comment by zadjii-msft** [comment]
Do we think there's any way to get this fixed without an OS-level update? We get roughly one bug filed every week about this particular bug 😕. It's not particularly painful, just more of an annoyance / polish issue.

**Comment by StephenLPeters** [comment]
I don't think that will be possible, and any fix will definitely not go down level. However I think your likely able to work around this issue, Can you provide more info about the layout of the app? how large is the xaml Island?

**Comment by zadjii-msft** [comment]
The island takes up the entire size of our HWND. When we're running with the tabs in the titlebar, there's a small HWND we place over part of the titlebar (for intercepting input, for dragging the window), but otherwise the entire window is the Terminal.

* [The page that this particular box is on](https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalSettingsEditor/Launch.xaml)
* is contained in [this NavView](https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalSettingsEditor/MainPage.xaml#L37)
* that page is then placed inside [this Grid](https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalApp/TerminalPage.xaml#L24)
* that page is the root content for our app

**Comment by JesseCol** [comment]
Hi @zadjii-msft , sorry about this bug.  I believe folks have worked around these types of issues in the past by iterating over the open popups and closing them:

`            foreach (Popup p in VisualTreeHelper.GetOpenPopupsForXamlRoot(this.XamlRoot))
            {
                p.IsOpen = false;
            }
`

In your case, you could do this in response to ScrollViewer.ViewChanging and when your top-level window gets a WM_MOVE.

Does that help at all?

**Comment by zadjii-msft** [comment]
For future readers - yes that workaround did work for me. For folks who want to do something similar in C++/WinRT, https://github.com/microsoft/terminal/commit/41e48012a is how I did the `ScrollViewer.ViewChanging` part of this for one of our pages. I'll still have to go through the rest and do the WM_MOVE thing, but this will get me unblocked for now.

**Comment by DHowett** [comment]
There being no activity on an issue means the issue _continues existing,_ not that it disappears!

**Comment by vgromfeld** [comment]
Another workaround, which works for me, is to change the vertical offset of the popups. Somehow, it forces the XAML runtime to re-compute the position of the popups.

```cpp
auto popups{ winrt::Windows::UI::Xaml::Media::VisualTreeHelper::GetOpenPopupsForXamlRoot(XamlRoot()) };
for (const auto& popup : popups)
{
    auto offset = popup.VerticalOffset();
    popup.VerticalOffset(offset + 10);
    popup.VerticalOffset(offset);
 }
```
